src/qt/optionsmodel.cpp

```diff
@@ -403,13 +391,21 @@ void OptionsModel::SetPruneEnabled(bool prune, bool force)
     settings.setValue("bPrune", prune);
     const int64_t prune_target_mib = PruneGBtoMiB(settings.value("nPruneSize").toInt());
     std::string prune_val = prune ? ToString(prune_target_mib) : "0";
+    auto maybe_adjust_dash_options = [](){
+        if (gArgs.GetArg("-prune", 0) > 0) {
+            gArgs.SoftSetBoolArg("-disablegovernance", true);
+            gArgs.SoftSetBoolArg("-txindex", false);
+        }
+    };
     if (force) {
         gArgs.ForceSetArg("-prune", prune_val);
+        maybe_adjust_dash_options();
```

Comment:
If we're using `ForceSetArg` for `-prune`, should we extend that to param interactions?

Reply:
```c++
    std::string prune_val = prune ? ToString(prune_target_mib) : "0";
    gArgs.ForceSetArg("-prune", prune_val));
    if (prune) {
        // Using ForceSetArg for parameter interactions to match with -prune set with ForceSetArg
        gArgs.ForceSetArg("-disablegovernance", true);
        gArgs.ForceSetArg("-txindex", false);
    }
```

Reply:
Makes sense I guess, pushed